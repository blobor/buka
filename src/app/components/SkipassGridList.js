import React, { Component } from 'react'
import SkipassCard from './SkipassCard'

class SkipassGridList extends Component {
  componentDidMount () {
    this.props.initialize()
  }

  render () {
    const { skipasses } = this.props

    return (
      <section className='skipasess__container'>
        {skipasses.map((skipass, index) => <SkipassCard key={index} skipass={skipass} />)}
      </section>
    )
  }
}

export default SkipassGridList
