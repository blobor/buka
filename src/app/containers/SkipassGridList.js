import { connect } from 'react-redux'
import { fetchStoredSkipasses, removeStoredSkipass } from '../actions/store-skipass'
import SkipassGridList from '../components/SkipassGridList'

const mapStateToProps = state => {
  return {
    skipasses: state.get('storedSkipasses').toJS()
  }
}

const mapDispatchToProps = dispatch => {
  return {
    initialize: () => {
      dispatch(fetchStoredSkipasses())
    },
    onRemoveSkipass: skipass => {
      dispatch(removeStoredSkipass(skipass))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SkipassGridList)
