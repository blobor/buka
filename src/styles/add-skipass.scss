.add-skipass {
  &__button {
    position: fixed;
    right: 10pt;
    bottom: 20pt;

    &--hidden {
      display: none !important;
    }
  }
}