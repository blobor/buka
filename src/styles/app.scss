@import "common/material-html";
@import "common/material-shadows";
@import "common/material-data-table";

@import "layout/header";
@import "layout/container";
@import "layout/footer";
@import "styles";
@import "search";
@import "add-skipass";
@import "lifts-table";
@import "skipass-info";
@import "skipass-grid-list";
@import "skipass-card";
@import "about";
