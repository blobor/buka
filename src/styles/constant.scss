$header-height: 64px;
$footer-height: 42pt;
