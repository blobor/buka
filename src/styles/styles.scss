@import "constant.scss";
@import "screen-sizes.scss";

html, body {
    font-family: 'Roboto', sans-serif;
}

.buka {
    display: flex;
    flex-direction: column;
    font-size: 14px;
    min-height: 100vh;

  &__container {
      display: flex;
      flex: 1 0 auto;
      flex-direction: column;
      align-items: center;
      box-sizing: border-box;
      padding: 0 10pt;
      margin: $header-height auto calc(#{$footer-height} + 5pt);
      width: 100%;
      max-width: 980px;
  }

  &__footer {
    flex-shrink: 0;
    position: fixed;
    left: 0;
    bottom: 0;
    width: 100%;
    height: $footer-height;
    background-color: #FFF;
  }
}

@media (max-width: $xsmall) {
  .buka__container {
    padding: 0 2pt;
  }
}

@media (max-width: $small) {
  .buka__container {
    padding: 0 5pt;
  }
}

.buka-skipass-number__form {
    width: 100%;
    display: flex;
    align-items: center;
    align-self: center;
    justify-content: center;
    flex-wrap: wrap;
}

.buka-skipass-number__input input {

  &::-ms-clear {
    display: none;
  }

  &:invalid {
    box-shadow: none;
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }
}

.buka-skipass-search__add-button--hidden {
  display: none !important;
}

.bottom-navigation {
  display: flex;
  justify-content: center;

  &__link {
    max-width: 168px;
    flex: 1;

    button {
      // workaround for FF
      width: 100%;
    }
  }
}
