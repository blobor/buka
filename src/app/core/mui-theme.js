import { fullWhite } from 'material-ui/styles/colors'
import { getMuiTheme as getTheme } from 'material-ui/styles'

export const getMuiTheme = userAgent => {
  return getTheme({
    userAgent: userAgent,
    textField: {
      hintColor: '#565656'
    },
    svgIcon: {
      color: fullWhite
    }
  })
}