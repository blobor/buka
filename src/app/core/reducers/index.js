import { combineReducers } from 'redux-immutable'
import app from './app'
import searchSkipass from './search-skipass'
import skipasses from './skipasses'

const rootReducer = combineReducers({
  app,
  searchSkipass,
  skipasses
})

export default rootReducer
