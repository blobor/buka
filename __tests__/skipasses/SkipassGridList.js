/* eslint-env jest */

import React from 'react'
import moment from 'moment'
import renderer from 'react-test-renderer'
import { MuiThemeProvider } from 'material-ui/styles'
import { getMuiTheme } from '../../src/app/core/mui-theme'
import SkipassGridList from '../../src/app/skipasses/SkipassGridList.component'

jest.mock('../../src/app/helpers/date', () => ({
  getAdoptedDateString: date => date.toISOString()
}))

describe('SkipassGridList', () => {
  it('should renders correctly', () => {
    const skipasses = [{
      ticketNumber: '01-1614 7133 5345 3531 0476-4',
      name: '5 днів СЕЗОН (ВСЕF)',
      purchaseDate: moment.utc('2015-03-09').toDate(),
      cardNumber: '01-2167-30-92545',
      lifts: [],
      isUnUsed: false,
      balance: 0,
      syncDate: moment.utc('2017-01-09').toDate(),
      updateDate: moment.utc('2017-01-09').toDate()
    }, {
      ticketNumber: '26-1614 7097 9217 9800 3595-2',
      name: '1 День (ДОР)',
      purchaseDate: moment.utc('2011-01-30').toDate(),
      cardNumber: '26-2167-19-35623',
      lifts: [],
      isUnUsed: false,
      balance: 0,
      syncDate: moment.utc('2017-01-09').toDate(),
      updateDate: moment.utc('2017-01-09').toDate()
    }]
    const onRemoveSkipass = jest.fn()
    const muiTheme = getMuiTheme()

    const tree = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <SkipassGridList skipasses={skipasses} onRemoveSkipass={onRemoveSkipass} />
      </MuiThemeProvider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
