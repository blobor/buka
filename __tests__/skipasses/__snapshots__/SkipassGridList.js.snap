exports[`SkipassGridList should renders correctly 1`] = `
<section
  className="skipasess__container">
  <article
    className="skipass-card md-shadow--2dp md-shadow--animated md-shadow--2dp-interactive skipass-card--zero-balance">
    <header
      className="skipass-card__header">
      <h5
        className="skipass-card__title">
        <a
          className="skipass-card__link"
          href="/skipasses/01-2167-30-92545"
          onClick={[Function]}
          style={Object {}}>
          5 днів СЕЗОН (ВСЕF)
        </a>
      </h5>
      <button
        className="skipass-card__close-button"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "display": "inline-block",
                "fill": "rgba(255, 255, 255, 1)",
                "height": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
                "width": 24,
              }
            }
            viewBox="0 0 24 24">
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
          </svg>
        </div>
      </button>
    </header>
    <a
      className="skipass-card__link skipass-card__main-container"
      href="/skipasses/01-2167-30-92545"
      onClick={[Function]}
      style={Object {}}>
      <h5
        className="skipass-card__card-number">
        01-2167-30-92545
      </h5>
      <section
        className="skipass-card__info-container">
        <time
          className="skipass-card__purchase-date"
          dateTime={2015-03-09T00:00:00.000Z}>
          2015-03-09T00:00:00.000Z
        </time>
      </section>
      <section
        className="skipass-card__balance-container">
        <h5
          className="skipass-card__balance-title">
          Balance
        </h5>
        <span
          className="skipass-card__balance">
          0
        </span>
      </section>
    </a>
  </article>
  <article
    className="skipass-card md-shadow--2dp md-shadow--animated md-shadow--2dp-interactive skipass-card--zero-balance">
    <header
      className="skipass-card__header">
      <h5
        className="skipass-card__title">
        <a
          className="skipass-card__link"
          href="/skipasses/26-2167-19-35623"
          onClick={[Function]}
          style={Object {}}>
          1 День (ДОР)
        </a>
      </h5>
      <button
        className="skipass-card__close-button"
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "border": 10,
            "borderRadius": 2,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": "inherit",
            "fontWeight": "inherit",
            "height": 36,
            "lineHeight": "36px",
            "margin": 0,
            "minWidth": 88,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "textAlign": "center",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "verticalAlign": null,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "display": "inline-block",
                "fill": "rgba(255, 255, 255, 1)",
                "height": 24,
                "marginLeft": 0,
                "marginRight": 0,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "verticalAlign": "middle",
                "width": 24,
              }
            }
            viewBox="0 0 24 24">
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z" />
          </svg>
        </div>
      </button>
    </header>
    <a
      className="skipass-card__link skipass-card__main-container"
      href="/skipasses/26-2167-19-35623"
      onClick={[Function]}
      style={Object {}}>
      <h5
        className="skipass-card__card-number">
        26-2167-19-35623
      </h5>
      <section
        className="skipass-card__info-container">
        <time
          className="skipass-card__purchase-date"
          dateTime={2011-01-30T00:00:00.000Z}>
          2011-01-30T00:00:00.000Z
        </time>
      </section>
      <section
        className="skipass-card__balance-container">
        <h5
          className="skipass-card__balance-title">
          Balance
        </h5>
        <span
          className="skipass-card__balance">
          0
        </span>
      </section>
    </a>
  </article>
</section>
`;
